### PR TITLE
Release candidate 4.5.0

### PR DESCRIPTION
**Pull request by alexharv074**
This branch merges these PRs for local testing with a view to releasing as Sceptre v4.5.0.

Screenshot 2024-06-03 at 10 49 42 am

## PR Checklist

- [ ] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [ ] All unit tests (`poetry run tox`) are passing.
- [ ] Used the same coding conventions as the rest of the project.
- [ ] The new code passes pre-commit validations (`poetry run pre-commit run --all-files`).
- [ ] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

File: tests/test_actions.py [hunk 1]
```diff
@@ -397,6 +399,19 @@ def test_launch_with_stack_that_failed_to_create(
         mock_create.assert_called_once_with()
         assert response == sentinel.launch_response
 
+    @patch("sceptre.plan.actions.StackActions.create")
+    @patch("sceptre.plan.actions.StackActions.delete")
+    @patch("sceptre.plan.actions.StackActions._get_status")
+    def test_launch_with_stack_that_does_not_exist(
+        self, mock_get_status, mock_delete, mock_create
+    ):
+        mock_get_status.return_value = "REVIEW_IN_PROGRESS"
```
Comment (ericabrauer) [inline tests/test_actions.py:408]: Are lines 408 and 383 swapped? 

```
380: def test_launch_with_stack_in_review_in_progress( ...
...
383: mock_get_status.side_effect = StackDoesNotExistError()
```

```
405:  def test_launch_with_stack_that_does_not_exist(
...
408: mock_get_status.return_value = "REVIEW_IN_PROGRESS"
```

**Review by ericabrauer — COMMENTED** [review]
Just had the one comment - looks great! Nice work!

File: tests/test_actions.py (hunk 1 above)
Comment (alexharv074) [reply tests/test_actions.py:408]: This is actually coming from one of the PRs merged into this integration branch, and it's just a quirk of how GitHub is rendering the diffs:

Screenshot 2024-06-12 at 11 45 03 am

**Comment by alexharv074** [comment]
Output is a bit inconsistent for change sets of type CREATE 

Screenshot 2024-07-06 at 4 48 55 pm

**Comment by alexharv074** [comment]
All in master now.

**Pull request closed without merging**